cadc-log/src/main/java/ca/nrc/cadc/log/LogControlServlet.java

```diff
@@ -359,29 +371,173 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     /**
      * Check for proper group membership.
      */
-    private void authorize(HttpServletRequest request, boolean readOnly) throws AccessControlException, TransientException {
-        if (authorizerClassName != null && groupAuthorizer == null) {
-            throw new RuntimeException("CONFIG: group authorizer was configured but failed to load: " + authorizerClassName);
+    private void authorize(HttpServletRequest request, boolean readOnly)
+        throws AccessControlException, TransientException {
+
+        // Track exceptions thrown
+        AccessControlException acException = null;
+
+        // Get the logControl properties if they exist.
+        PropertiesReader propertiesReader = getLogControlProperties();
+
+        // first check if request user matches authorized config file users
+        Set<Principal> authorizedUsers = getAuthorizedUserPrincipals(propertiesReader);
+        if (authorizedUsers.isEmpty()) {
+            logger.debug("Authorized users not configured");
+        } else {
+            try {
+                doUserCheck(request, authorizedUsers);
+                return;
+            } catch (Exception e) {
+                if (e instanceof AccessControlException) {
+                    logger.warn("User authorization failed: " + e.getMessage());
+                    acException = (AccessControlException) e;
+                }
+            }
         }
 
-        if (groupAuthorizer == null) {
-            logger.warn("Authorization not configured, log control is public.");
-            return;
+        // Check for groups configured in servlet init or properties file.
+        Set<GroupURI> groupUris = getAuthorizedGroupUris(propertiesReader);
+
+        // If no groups configured, and no users configured, then public access and return.
+        // If no groups configured, and a user exception was previously thrown,
+        // throw the user AccessControlException.
+        if (groupUris.isEmpty() && accessGroup == null) {
+            if (acException != null) {
+                throw acException;
+            } else {
+                logger.info("Authorization not configured, log control is public.");
+                return;
+            }
         }
+        logger.debug("Configured GroupURI's: " + groupUris);
 
+        // Augment the calling subject.
```

Comment:
If you have a non-augmented subject from above, I'm not sure you need an augmented one here. You have a DN for the caller, which is enough to do a group check. 

I also do not see any call to cred to load a client proxy cert so I don't see how the calls to /ac are going to work. That has to happen once before looping over groups.

Reply:
Added a call to cred to get a client proxy cert.